DNN Platform/Library/Entities/Modules/ModuleController.cs

```diff
@@ -1584,6 +1584,26 @@ public ArrayList GetModulesByDefinition(int portalID, string definitionName)
         }
 
         /// <summary>
+        /// Gets the modules by DesktopModuleId.
+        /// </summary>
+        /// <param name="desktopModuleId">The Desktop Module Id.</param>
+        /// <returns>module collection</returns>
+        public ArrayList GetModulesByDesktopModuleId(int desktopModuleId)
+        {
+            var moduleDefinitions = ModuleDefinitionController.GetModuleDefinitionsByDesktopModuleID(desktopModuleId);
+            var modules = new ArrayList();
+            foreach (var moduleDefinition in moduleDefinitions)
+            {
+                var portals = PortalController.Instance.GetPortals();
+                foreach (PortalInfo portal in portals)
+                {
+                    modules.AddRange(GetModulesByDefinition(portal.PortalID, moduleDefinition.Key));

```

Comment:
the dictionary's key is module definition's friendly name but not definition name(they are same value mostly, but it could be different), need use the definition name here.